### ISSUE TITLE
Traceback from invalid ISOdatetime point not swallowed.

### ISSUE THREAD

**Issue by wxtim** [cylc/cylc-flow#4630]
**Describe the bug**
```
$ cylc play clock-trigger-tutorial -n
...
ERROR - (type=cycle point) [scheduling]initial cycle point = TODO
    Traceback (most recent call last):
      File "~/cylc-flow/cylc/flow/parsec/validate.py", line 789, in coerce_cycle_point
        TimePointParser().parse(value)
      File "~/isodatetime/metomi/isodatetime/parsers.py", line 240, in parse
        date_info, time_info, parsed_expr = self.get_info(timepoint_string)
      File "~/isodatetime/metomi/isodatetime/parsers.py", line 434, in get_info
        keys, date_info = self.get_date_info(date)
      File "~/isodatetime/metomi/isodatetime/parsers.py", line 394, in get_date_info
        raise ISO8601SyntaxError("date", date_string)
    metomi.isodatetime.exceptions.ISO8601SyntaxError: Invalid ISO 8601 date representation: TODO
    During handling of the above exception, another exception occurred:
    Traceback (most recent call last):
      File "~/cylc-flow/cylc/flow/scheduler.py", line 657, in run
        await self.configure()
      File "~/cylc-flow/cylc/flow/scheduler.py", line 442, in configure
        self.load_flow_file()
      File "~/cylc-flow/cylc/flow/scheduler.py", line 1098, in load_flow_file
        self.config = WorkflowConfig(
      File "~/cylc-flow/cylc/flow/config.py", line 253, in __init__
        self.pcfg = RawWorkflowConfig(
      File "~/cylc-flow/cylc/flow/cfgspec/workflow.py", line 1981, in __init__
        self.loadcfg(fpath, "workflow definition")
      File "~/cylc-flow/cylc/flow/parsec/config.py", line 76, in loadcfg
        self.validate(sparse)
      File "~/cylc-flow/cylc/flow/parsec/config.py", line 86, in validate
        return self.validator(sparse, self.spec)
      File "~/cylc-flow/cylc/flow/parsec/validate.py", line 1044, in cylc_config_validate
        return CylcConfigValidator().validate(cfg_root, spec_root)
      File "~/cylc-flow/cylc/flow/parsec/validate.py", line 212, in validate
        cfg[key] = self.coercers[specval.vdr](value, keys + [key])
      File "~/cylc-flow/cylc/flow/parsec/validate.py", line 791, in coerce_cycle_point
        raise IllegalValueError('cycle point', keys, value)
    cylc.flow.parsec.exceptions.IllegalValueError: (type=cycle point) [scheduling]initial cycle point = TODO
CRITICAL - Workflow shutting down - (type=cycle point) [scheduling]initial cycle point = TODO
```

**Steps to reproduce the bug**
Create, install a workflow with an invalid string for `[scheduling]initial cycle point`
Error when you `cylc play`.

**Expected behavior**
Traceback should return a one line error when run at standard logging level: It should look rather like what happens if you set `initial cycle point = `

**Additional context**
You should probably check whether other fields, such as final cycle point have the same problem.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by oliver-sanders** [comment]
(should catch this error and re-raise a more friendly `UserInputError`)

**Comment by wxtim** [comment]
It looks like this happens with any config item not the right parsec type.

**Comment by oliver-sanders** [comment]
May need to handle ParsecError the same way as CylcError here:

[de8dea38d1](https://github.com/cylc/cylc-flow/pull/4168/files#diff-90711b005fe27b9f894e6cd09f08816d847dcf853518894c6b7e2139784cbcb8R1622)

**Comment by oliver-sanders** [comment]
FYI, can test integration style:

```python
async with run(schd) as log:
    pass
assert log.record_tuples == [
   ...
]
```

**Comment by wxtim** [comment]
Not sure that works, because this problem occurs in the `configure` method, which is called before the scheduler proper starts.

**Comment by oliver-sanders** [comment]
Errors during configuration are handled by `handle_exception`:

https://github.com/cylc/cylc-flow/blob/6e2021c45ed5bb99e2fc73ba51aa764277b00bd5/cylc/flow/scheduler.py#L655-L661

Then `handle_exception` calls shutdown:

https://github.com/cylc/cylc-flow/blob/6e2021c45ed5bb99e2fc73ba51aa764277b00bd5/cylc/flow/scheduler.py#L1966-L1975

And shutdown logs the traceback:

https://github.com/cylc/cylc-flow/blob/6e2021c45ed5bb99e2fc73ba51aa764277b00bd5/cylc/flow/scheduler.py#L1656-L1675

**Comment by oliver-sanders** [comment]
This is the same behaviour as Cylc 7 BTW.

**Comment by oliver-sanders** [comment]
Downgrading from bug since this is long standing behaviour working as intended, could be better though...

**Comment by oliver-sanders** [comment]
We want to swallow errors from the config load but not from anywhere else (because if a `Parsec` error were to occur anywhere besides config load that would be a real problem).

One option is to use a special exception as a messenger class something like this:

```python
class ConfigError: pass  # note doesn't inherit from CylcError

...

try:
    # load config
except ParsecError as exc:
    raise ConfigError(as exc)

...

async def handle_exception(exc)
    await shutdown(exc)
    if isinstance(exc, ConfigError):
        orig_exc = exc.args[0]  # or whatever
        log_nicely(orig_exc)  # (see cylc.flow.terminal)
    raise from None

```

**Comment by MetRonnie** [comment]
I get traceback for `IllegalItemError`s as well as `IllegalValueError`s, but strangely not if the final cycle point is dodgy - I just get the one liner:
```
ERROR - Workflow shutting down - CylcTimeSyntaxError: '2000-03-021T00Z': not a valid cylc-shorthand or full ISO 8601 date representation
```
Both initial and stop after cycle point mistakes give traceback, just not final cycle point for some reason

**Comment by MetRonnie** [comment]
If you use `--icp` and make a mistake there, you just get the one liner
```
ERROR - Workflow shutting down - WorkflowConfigError: Invalid ISO 8601 date representation: 2000-01-021
```
However a mistake in `--stopcp` gives traceback!

**Comment by MetRonnie** [comment]
See also the discussion at https://github.com/cylc/cylc-flow/issues/4216

**Comment by MetRonnie** [comment]
Closing as duplicate of #4216
